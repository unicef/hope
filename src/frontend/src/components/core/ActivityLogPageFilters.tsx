import ViewModuleRoundedIcon from '@mui/icons-material/ViewModuleRounded';
import { Grid, MenuItem } from '@mui/material';
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { useLocation, useNavigate } from 'react-router-dom';
import { AssigneeAutocomplete } from '@shared/autocompletes/AssigneeAutocomplete';
import { createHandleApplyFilterChange } from '@utils/utils';
import { FiltersSection } from './FiltersSection';
import { SearchTextField } from './SearchTextField';
import { SelectFilter } from './SelectFilter';

interface ActivityLogPageFiltersProps {
  filter;
  setFilter: (filter) => void;
  initialFilter;
  appliedFilter;
  setAppliedFilter: (filter) => void;
}
export function ActivityLogPageFilters({
  filter,
  setFilter,
  initialFilter,
  appliedFilter,
  setAppliedFilter,
}: ActivityLogPageFiltersProps): React.ReactElement {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();

  const { handleFilterChange, applyFilterChanges, clearFilter } =
    createHandleApplyFilterChange(
      initialFilter,
      navigate,
      location,
      filter,
      setFilter,
      appliedFilter,
      setAppliedFilter,
    );

  const handleApplyFilter = (): void => {
    applyFilterChanges();
  };

  const handleClearFilter = (): void => {
    clearFilter();
  };

  const modules = {
    program: 'Programme',
    household: 'Household',
    individual: 'Individual',
    grievanceticket: 'Grievance Tickets',
    paymentverificationplan: 'Payment Plan Payment Verification',
    targetpopulation: 'Target Population',
    registrationdataimport: 'Registration Data Import',
  };
  return (
    <FiltersSection
      clearHandler={handleClearFilter}
      applyHandler={handleApplyFilter}
    >
      <Grid container alignItems="center" spacing={3}>
        <Grid item xs={3}>
          <SearchTextField
            label={t('Search')}
            value={filter.search}
            onChange={(e) => handleFilterChange('search', e.target.value)}
            data-cy="filters-search"
          />
        </Grid>
        <Grid item xs={3}>
          <SelectFilter
            onChange={(e) => handleFilterChange('module', e.target.value)}
            label={t('Module')}
            value={filter.module}
            icon={<ViewModuleRoundedIcon />}
            SelectDisplayProps={{
              'data-cy': 'filters-residence-status',
            }}
            MenuProps={{
              'data-cy': 'filters-residence-status-options',
            }}
          >
            {Object.entries(modules)
              .sort()
              .map(([key, value]) => (
                <MenuItem key={key} value={key}>
                  {value}
                </MenuItem>
              ))}
          </SelectFilter>
        </Grid>
        <Grid item xs={3}>
          <AssigneeAutocomplete
            label={t('User')}
            filter={filter}
            name="userId"
            value={filter.userId}
            setFilter={setFilter}
            initialFilter={initialFilter}
            appliedFilter={appliedFilter}
            setAppliedFilter={setAppliedFilter}
          />
        </Grid>
      </Grid>
    </FiltersSection>
  );
}
