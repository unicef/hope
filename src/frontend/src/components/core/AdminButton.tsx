import ArrowCircleRightIcon from '@mui/icons-material/ArrowCircleRight';
import * as React from 'react';


interface AdminButtonProps {
    adminUrl: string;
    [key: string]: any;
}

export const AdminButton: React.FC<AdminButtonProps> = ({
    adminUrl,
    ...otherProps
}) => {
    if (adminUrl) {
        return <a href={adminUrl}><ArrowCircleRightIcon color="primary" {...otherProps}/></a>;
    }
    return null;
};
