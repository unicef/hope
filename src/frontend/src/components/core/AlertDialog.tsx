import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import * as React from 'react';
import { useEffect } from 'react';
import { useTranslation } from 'react-i18next';

export function AlertDialog({
  message,
  show,
}: {
  message: string;
  show: boolean;
}): React.ReactElement {
  const { t } = useTranslation();
  const [open, setOpen] = React.useState(false);
  const [oldShow, setOldShow] = React.useState(false);
  useEffect(() => {
    if (!oldShow && show) {
      setOpen(true);
    }
    setOldShow(show);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [show]);

  const handleClose = (): void => {
    setOpen(false);
  };

  return (
    <div>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{t('Error')}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            {message}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color="primary" autoFocus>
            OK
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
