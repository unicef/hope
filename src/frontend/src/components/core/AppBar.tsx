import { BusinessAreaSelect } from '@containers/BusinessAreaSelect';
import { GlobalProgramSelect } from '@containers/GlobalProgramSelect';
import { UserProfileMenu } from '@containers/UserProfileMenu';
import { useCachedMe } from '@hooks/useCachedMe';
import MenuIcon from '@mui/icons-material/Menu';
import TextsmsIcon from '@mui/icons-material/Textsms';
import { Box, Button } from '@mui/material';
import MuiAppBar from '@mui/material/AppBar';
import IconButton from '@mui/material/IconButton';
import Toolbar from '@mui/material/Toolbar';
import { styled } from '@mui/system';
import * as React from 'react';
import { MiśTheme, theme } from 'src/theme';

const StyledToolbar = styled(Toolbar)(() => ({
  display: 'flex',
  justifyContent: 'space-between',
}));

const StyledLink = styled('a')({
  textDecoration: 'none',
  color: '#e3e6e7',
});

interface AppBarProps {
  open: boolean;
}

interface StyledAppBarProps extends AppBarProps {
  theme: MiśTheme;
}
const StyledAppBar = styled(MuiAppBar)<StyledAppBarProps>(
  ({ theme: muiTheme, open }) => ({
    position: 'fixed',
    top: 0,
    zIndex: muiTheme.zIndex.drawer + 1,
    backgroundColor: muiTheme.palette.secondary.main,
    transition: muiTheme.transitions.create(['width', 'margin'], {
      easing: muiTheme.transitions.easing.sharp,
      duration: muiTheme.transitions.duration.leavingScreen,
    }),
    ...(open && {
      marginLeft: '270px',
      width: 'calc(100% - 270px)',
      transition: muiTheme.transitions.create(['width', 'margin'], {
        easing: muiTheme.transitions.easing.sharp,
        duration: muiTheme.transitions.duration.enteringScreen,
      }),
    }),
    ...(!open && {
      marginLeft: '0',
      width: '100%',
      transition: muiTheme.transitions.create(['width', 'margin'], {
        easing: muiTheme.transitions.easing.sharp,
        duration: muiTheme.transitions.duration.leavingScreen,
      }),
    }),
  }),
);

const StyledIconButton = styled(IconButton)<AppBarProps>(({ open }) => ({
  marginRight: 36,
  ...(open && {
    display: 'none',
  }),
}));

export const AppBar = ({ open, handleDrawerOpen }): React.ReactElement => {
  const { data: meData, loading: meLoading } = useCachedMe();
  const servicenow = `https://unicef.service-now.com/cc?id=sc_cat_item&sys_id=762ae3128747d91021cb670a0cbb35a7&HOPE - ${
    window.location.pathname.split('/')[2]
  }&Workspace: ${window.location.pathname.split('/')[1]} \n Url: ${
    window.location.href
  }`;

  if (meLoading) {
    return null;
  }
  return (
    <StyledAppBar theme={theme} open={open}>
      <StyledToolbar>
        <Box display="flex" alignItems="center" justifyContent="center">
          <Box ml={1}>
            <StyledIconButton
              edge="start"
              color="inherit"
              aria-label="open drawer"
              onClick={handleDrawerOpen}
              open={open}
            >
              <MenuIcon />
            </StyledIconButton>
          </Box>
          <Box display="flex" alignItems="center">
            <Box ml={6} data-cy="business-area-container">
              <BusinessAreaSelect />
            </Box>
            <Box ml={6} data-cy="global-program-filter-container">
              <GlobalProgramSelect />
            </Box>
          </Box>
        </Box>
        <Box display="flex" justifyContent="flex-end">
          <Button startIcon={<TextsmsIcon style={{ color: '#e3e6e7' }} />}>
            <StyledLink target="_blank" href={servicenow}>
              Support
            </StyledLink>
          </Button>
          <UserProfileMenu meData={meData} />
        </Box>
      </StyledToolbar>
    </StyledAppBar>
  );
};
