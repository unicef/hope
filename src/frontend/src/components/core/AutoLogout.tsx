import * as React from 'react';
import { useCallback, useEffect } from 'react';
import { useIdleTimer } from 'react-idle-timer';
import { AUTO_LOGOUT_MILLIS } from '../../config';

export function AutoLogout(): React.ReactElement {
  const handleOnIdle = (): void => {
    if (!localStorage.getItem('AUTHENTICATED')) {
      return;
    }
    window.location.assign('/api/logout');
    localStorage.removeItem('AUTHENTICATED');
  };

  //eslint-disable-next-line
  const { reset } = useIdleTimer({
    timeout: AUTO_LOGOUT_MILLIS,
    onIdle: handleOnIdle,
    debounce: 500,
  });

  const handleOnAction = useCallback((): void => {
    reset();
  }, [reset]);

  useEffect(() => {
    window.addEventListener('click', handleOnAction);
    window.addEventListener('mousemove', handleOnAction);

    return () => {
      window.removeEventListener('click', handleOnAction);
      window.removeEventListener('mousemove', handleOnAction);
    };
  }, [handleOnAction]);

  return <></>;
}
