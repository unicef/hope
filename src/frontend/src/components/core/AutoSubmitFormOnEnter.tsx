import { useFormikContext } from 'formik';
import * as React from 'react';
import { useEffect } from 'react';

export function AutoSubmitFormOnEnter(): React.ReactElement {
  const { submitForm } = useFormikContext();
  useEffect(() => {
    const handleEnter = (e: KeyboardEvent): void => {
      if (e.key === 'Enter') {
        submitForm();
      }
    };
    window.addEventListener('keypress', handleEnter);
    return () => {
      window.removeEventListener('keypress', handleEnter);
    };
  }, [submitForm]);
  return <></>;
}
