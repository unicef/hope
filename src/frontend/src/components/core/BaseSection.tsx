import { Box, Typography } from '@mui/material';
import { ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { GreyText } from './GreyText';

const PaperContainer = styled(Box)`
  display: flex;
  padding: ${({ theme }) => theme.spacing(3)} ${({ theme }) => theme.spacing(4)};
  flex-direction: column;
  background-color: #fff;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  width: 100%;
`;

const BoxContainer = styled(Box)`
  display: flex;
  padding: ${({ theme }) => theme.spacing(3)} ${({ theme }) => theme.spacing(4)};
  flex-direction: column;
  width: 100%;
`;

interface BaseSectionProps {
  children?: ReactElement | ReactElement[];
  buttons?: ReactElement;
  title?: string | ReactElement;
  description?: string;
  p?: number;
  noPaper?: boolean;
  tabs?: ReactElement;
  stepper?: ReactElement;
}

export const BaseSection = ({
  children = <></>,
  buttons,
  title = '',
  description,
  p = 3,
  noPaper = false,
  tabs = null,
  stepper = null,
}: BaseSectionProps): React.ReactElement => {
  const { t } = useTranslation();
  const Container = noPaper ? BoxContainer : PaperContainer;

  return (
    <Container>
      <Box p={p}>
        {stepper && <Box mb={2}>{stepper}</Box>}
        <Box
          p={p}
          display="flex"
          justifyContent="space-between"
          alignItems="center"
        >
          <Box>
            {typeof title === 'string' ? (
              <Typography data-cy="title" variant="h6">
                {t(title)}
              </Typography>
            ) : (
              title
            )}
          </Box>
          {buttons}
        </Box>
        {tabs && <Box mb={2}>{tabs}</Box>}
        {description && (
          <Box pl={p} mb={2}>
            <GreyText data-cy="description">{description}</GreyText>
          </Box>
        )}
        {children}
      </Box>
    </Container>
  );
};
