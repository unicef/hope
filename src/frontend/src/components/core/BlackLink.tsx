import * as React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

interface StyledLinkProps {
  fullWidth?: boolean;
  newTab?: boolean;
}

const StyledLink = styled(Link)<StyledLinkProps>`
  color: #000;
  max-width: ${(props) => (props.fullWidth ? '100%' : '200px')}
  overflow-wrap: break-word;
`;

const StyledLinkA = styled.a<StyledLinkProps>`
  color: #000;
  max-width: ${(props) => (props.fullWidth ? '100%' : '200px')}
  overflow-wrap: break-word;
`;

export const BlackLink = ({
  newTab = false,
  to = '/',
  ...props
}): React.ReactElement => {
  return newTab ? (
    <StyledLinkA
      {...props}
      href={to}
      target="_blank"
      rel="noopener noreferrer"
      onClick={(e) => e.stopPropagation()}
    />
  ) : (
    <StyledLink {...props} to={to} onClick={(e) => e.stopPropagation()} />
  );
};
