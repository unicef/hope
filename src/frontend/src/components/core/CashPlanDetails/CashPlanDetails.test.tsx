import * as React from 'react';
import { render } from '../../../testUtils/testUtils';
import { fakeCashPlan } from '../../../../fixtures/payments/fakeCashPlan';
import { CashPlanDetails } from '.';
import { fakeBaseUrl } from '../../../../fixtures/core/fakeBaseUrl';

describe('components/core/CashPlanDetails', () => {
  it('should render', () => {
    const { container } = render(
      <CashPlanDetails baseUrl={fakeBaseUrl} cashPlan={fakeCashPlan} />,
    );
    expect(container).toMatchSnapshot();
  });
});
