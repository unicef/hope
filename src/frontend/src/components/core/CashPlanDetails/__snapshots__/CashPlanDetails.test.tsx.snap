// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/core/CashPlanDetails should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-beySPh ehchJK css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-qfdd9t"
      style="width: 100%;"
    >
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <div
          class="sc-iBdnpw fIJfic"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
          >
            Payment Plan Details
          </h6>
        </div>
        <div
          class="sc-eDLKkx dztDbI"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 css-zow5z4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  Status
                </span>
                <div
                  data-cy="label-Status"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    <div
                      class="sc-jTQCzO eAkjXM"
                    >
                      <div
                        class="sc-gLLuof bfnybE status-box-container"
                        data-cy="status-container"
                      >
                        DISTRIBUTION COMPLETED
                      </div>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  Plan Start Date
                </span>
                <div
                  data-cy="label-Plan Start Date"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    <time
                      datetime="1594703053000"
                    >
                      14 Jul 2020
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  Plan End Date
                </span>
                <div
                  data-cy="label-Plan End Date"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    <time
                      datetime="1679029453000"
                    >
                      17 Mar 2023
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  Payment Plan Name
                </span>
                <div
                  data-cy="label-Payment Plan Name"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    Military citizen until amount.
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  delivery type
                </span>
                <div
                  data-cy="label-delivery type"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    Voucher
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  assistance through
                </span>
                <div
                  data-cy="label-assistance through"
                >
                  -
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  dispertion date
                </span>
                <div
                  data-cy="label-dispertion date"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    <time
                      datetime="1639458253000"
                    >
                      14 Dec 2021
                    </time>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  fc id
                </span>
                <div
                  data-cy="label-fc id"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    62947144
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  dp id
                </span>
                <div
                  data-cy="label-dp id"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    27371694
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div>
                <span
                  class="sc-dmyCSP edirvw"
                  color="textSecondary"
                >
                  Target population(s)
                </span>
                <div
                  data-cy="label-Target population(s)"
                >
                  <span
                    class="sc-hLQSwg hXyiIZ"
                    color="textSecondary"
                  >
                    <span>
                      <a
                        class="sc-guDLey ibjlhn"
                        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/VGFyZ2V0UG9wdWxhdGlvbk5vZGU6ZGRlODE4MDUtMWQ0Ny00YzY0LTgzYTMtMzQ3NWEzMGVmZTJk"
                      >
                        Floor anyone remain play most.
                      </a>
                       
                    </span>
                    <span>
                      <a
                        class="sc-guDLey ibjlhn"
                        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/VGFyZ2V0UG9wdWxhdGlvbk5vZGU6MTdlM2VmMjAtNjczMy00ZDNhLWFiMDUtZjVkOTgzZjViMjI1"
                      >
                        Former center score fire role.
                      </a>
                       
                    </span>
                    <span>
                      <a
                        class="sc-guDLey ibjlhn"
                        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/VGFyZ2V0UG9wdWxhdGlvbk5vZGU6MjRkMTdhODktMjg4OC00NDI0LTgxMDAtYmU3MTZjOWFkNDk3"
                      >
                        Stock before maybe campaign blue history recently.
                      </a>
                       
                    </span>
                    <span>
                      <a
                        class="sc-guDLey ibjlhn"
                        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/VGFyZ2V0UG9wdWxhdGlvbk5vZGU6NzNiNTFhNjctZmRmNy00ODkwLWIyOWItMjhhYzk4NTc4OGRk"
                      >
                        Eight affect late place.
                      </a>
                       
                    </span>
                    <span>
                      <a
                        class="sc-guDLey ibjlhn"
                        href="/afghanistan/programs/UHJvZ3JhbU5vZGU6NmI1OTE1OTktYmNhNy00NDVhLWJmZjItYWU5MTUyMjMxZGFm/target-population/VGFyZ2V0UG9wdWxhdGlvbk5vZGU6ZjZkZjc2MDctMzMyZi00NDYwLWJiMzgtMTZiZGM4MGVkNTkz"
                      >
                        Seek development woman once worker.
                      </a>
                       
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-fsYfdN eoQjuz"
          >
            <div>
              <span
                class="sc-dmyCSP edirvw"
                color="textSecondary"
              >
                Total Number of Households
              </span>
              <div
                data-cy="label-Total Number of Households"
              >
                <span
                  class="sc-hLQSwg hXyiIZ"
                  color="textSecondary"
                >
                  <div
                    class="sc-qZrbh hnKrjX"
                  >
                    5
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
